src/schemas/ProgramDonorSummary/resolvers/summaryEntriesAndStats.ts

```diff
@@ -259,6 +260,79 @@ const programDonorSummaryEntriesAndStatsResolver: (
       const openAccessStatusQuery = esb.boolQuery().should(shouldQueries);
       queries.push(openAccessStatusQuery);
     }
+
+    if (field === EsDonorDocumentField.rnaRegisteredSample && filter.values.length > 0) {
+      const sampleQueries: Query[] = [];
+      for (const value of filter.values) {
+        switch (value) {
+          case RnaFilterStatus.NO_DATA:
+            const mustQueries: Query[] = [];
+            mustQueries.push(esb.rangeQuery(EsDonorDocumentField.rnaRegisteredNormalSamples).lte(0));
+            mustQueries.push(esb.rangeQuery(EsDonorDocumentField.rnaRegisteredTumourSamples).lte(0));
+            sampleQueries.push(esb.boolQuery().must(mustQueries));
+            break;
+          case RnaFilterStatus.DATA_SUBMITTED:
+            const shouldQueries: Query[] = [];
+            shouldQueries.push(esb.rangeQuery(EsDonorDocumentField.rnaRegisteredNormalSamples).gte(1));
+            shouldQueries.push(esb.rangeQuery(EsDonorDocumentField.rnaRegisteredTumourSamples).gte(1));
+            sampleQueries.push(esb.boolQuery().should(shouldQueries));
+            break;
+        }
+      }
+      const boolQuery = esb.boolQuery().should(sampleQueries);
+      queries.push(boolQuery);
+    }
```

Comment:
looks like this code is repeated for each field type (`rnaRegisteredSample`, `rnaPublishedNormalAnalysis`, `rnaAlignmentStatus`). probably want to write it out as a function that gets called in each case. easier to maintain changes consistently over each type.